uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/PublicHoliday.php

```diff
@@ -8,4 +8,189 @@
+            $message .= ' ' . $ex->getMessage();
+            CRM_Core_Session::setStatus($message, 'Error', 'error');
+        }
+
+        $url = CRM_Utils_System::url('civicrm/admin/leaveandabsences/public_holidays', 'reset=1&action=browse');
+        $session = CRM_Core_Session::singleton();
+        $session->replaceUserContext($url);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDefaultEntity() {
+    return 'PublicHoliday';
+  }
+
+  /**
+   * Adds fields to the Form.
+   */
+  private function addFields() {
+    $this->add(
+      'text',
+      'title',
+      ts('Title'),
+      $this->getDAOFieldAttributes('title'),
+      true
+    );
+    $this->add(
+      'datepicker',
+      'date',
+      ts('Date'),
+      $this->getDAOFieldAttributes('date'),
+      true,
+      ['time' => false]
+    );
+    $this->add(
+      'checkbox',
+      'is_active',
+      ts('Enabled')
+    );
+  }
+
+  /**
+   * Return an array containing attributes of given field.
+   * 
+   * @param string $field
+   * @return array|null
+   */
+  private function getDAOFieldAttributes($field) {
+    $dao = 'CRM_HRLeaveAndAbsences_DAO_PublicHoliday';
+    return CRM_Core_DAO::getAttribute($dao, $field);
+  }
+
+  /**
+   * Add validation rules for the fields on this form.
+   */
+  private function addFieldsRules() {
+    $this->addFormRule([$this, 'formRules']);
+  }
+
+  /**
+   * Execute validations concerning all the form fields.
+   *
+   * Different from regular field rules, Form Rules can be used for validations
+   * that depends on multiple fields,(for example, if field X is not empty,
+   * then Y is required). Here, we use it to validate the dates, because start
+   * date can't be greater or equal to end date.
+   *
+   * @param array $values An array containing all the form's fields values
+   *
+   * @return array|bool Returns true if form is valid. Otherwise, an
+   *                    array containing all the validation errors is returned.
+   */
+  public function formRules($values) {
+    $errors = [];
+    $this->validateDate($values, $errors);
+    return empty($errors) ? true : $errors;
+  }
+
+  /**
+   * Validates date.
+   *
+   * @param array $values An array containing all the form's fields values
+   * @param array $errors A reference to the errors array where errors will be
+   *                      added if dates are invalid
+   */
+  private function validateDate($values, &$errors) {
+    if (empty($values['date'])) {
+      return;
+    }
+
+    $dateIsValid = CRM_HRLeaveAndAbsences_Validator_Date::isValid($values['date'], 'Y-m-d');
+    if(!$dateIsValid) {
+      $errors['date'] = ts('Date should be a valid date');
+    }
+  }
+
+  private function getAvailableButtons() {
+    $buttons = [
+      [ 'type' => 'next', 'name' => ts('Save'), 'isDefault' => true ],
+      [ 'type' => 'cancel', 'name' => ts('Cancel') ],
+      [ 'type' => 'delete', 'name' => ts('Delete') ],
+    ];
```

Comment:
It's not stated on the ticket description, but can you change this so the Delete button is only available when editing an existing public holiday?